packages/strapi-versioning-plugin/src/middlewares/versions-middleware/index.ts

```diff
@@ -0,0 +1,22 @@
+const versioningService =
+  global.strapi.plugins["versioning-plugin"].services.versioning;
+
+module.exports = () => {
+  return {
+    initialize() {
+      global.strapi.app.use(async (ctx, next) => {
+        await next();
+        const properMethods = ["PUT", "POST"];
+        if (
+          properMethods.includes(ctx.response.request.method) &&
+          versioningService.isModelExists(ctx) &&
+          ctx.response.message == "OK"
+        ) {
+          const uid = versioningService.findUid(ctx);
+          const updatedData = await global.strapi.db.query(uid).find();
+          versioningService.saveDataInDB(updatedData, uid, ctx.response.body);
```

Comment:
You are saving body? without parse, without validation?

Reply:
validation is above in if statement, parsing in next app step